### PR TITLE
feat(auth): add change log

### PR DESCRIPTION
**Pull request by erinleigh90**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/v6-change-log/index.mdx [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Changes from v5 to v6',
+  description: '',
+  platforms: [
+    'android',
```
Comment (reesscot) [inline src/pages/[platform]/build-a-backend/auth/v6-change-log/index.mdx:7]: this should only show on the web platforms, right?

File: src/pages/[platform]/build-a-backend/auth/v6-change-log/index.mdx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
++    fetchUserAttributes: () => Promise<FetchUserAttributesOutput>;
+
+// Deprecated: use fetchUserAttributes() output.email_verified and output.phone_number_verified
+-    Auth.verifiedContact: (user: CognitoUser | any) => Promise<any>;
+     
+-    Auth.currentUserPoolUser: (params?: CurrentUserOpts) => Promise<CognitoUser | any>;
++    getCurrentUser: () => Promise<GetCurrentUserOutput>;
+
+-    Auth.currentAuthenticatedUser: (params?: CurrentUserOpts) => Promise<CognitoUser | any>;
++    getCurrentUser: () => Promise<GetCurrentUserOutput>;
```
Comment (calebpollman) [inline src/pages/[platform]/build-a-backend/auth/v6-change-log/index.mdx:95]: `Auth.currentAuthenticatedUser` returned additional data that is not provided by `GetCurrentUserOutput`, for example session related information. Can we please ensure that **all** output changes are accounted for? Also to match the rest of these examples, `Auth.currentAuthenticatedUser` should be noted as deprecated right?

File: src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+```
+
+## Auth.confirmSignUp
+
+```diff
+-  import { Auth } from 'aws-amplify';
++  import { confirmSignUp } from 'aws-amplify/auth';
+
+-  Auth.confirmSignUp: (username: string, code: string, options?: ConfirmSignUpOptions) => Promise<any>;
++  signUp: (input: SignUpInput) => Promise<SignUpOutput>;
```
Comment (israx) [inline src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx:45]: should be `confirmSignUp`

File: src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+
+```diff
+-  import { Auth } from 'aws-amplify';
++  import { signIn } from 'aws-amplify/auth';
+
+-  Auth.signIn: (usernameOrSignInOpts: string | SignInOpts, pw?: string, clientMetadata?: ClientMetaData) => Promise<CognitoUser | any>;
++  signIn: (input: SignInInput) => Promise<SignInOutput>;
+```
+
+## Auth.federatedSignIn
```
Comment (israx) [inline src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx:68]: I think it is worth to mention that the usage of `federate with identity pools` using this API is deprecated and pointing customers to this guide instead: https://docs.amplify.aws/react/build-a-backend/auth/advanced-workflows/#identity-pool-federation

File: src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+## Auth.currentUserPoolUser (DEPRECATED)
+
+This API has been deprecated: existing use cases can be migrated to a combination of the `getCurrentUser` and `fetchAuthSession` API's.
+
+```diff
+-  import { Auth } from 'aws-amplify';
++  import { getCurrentUser } from 'aws-amplify/auth';
+
+-  Auth.currentUserPoolUser: (params?: CurrentUserOpts) => Promise<CognitoUser | any>;
++  getCurrentUser: () => Promise<GetCurrentUserOutput>;
```
Comment (israx) [inline src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx:238]: The example for `currentAuthenticatedUser` suggests calling `getCurrentUser` and `fetchAuthSession`, probably is something that we can do here too

File: src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+
+```diff
+-  import { Auth } from 'aws-amplify';
++  import { fetchAuthSession } from 'aws-amplify/auth';
+
+-  Auth.currentSession: () => Promise<CognitoUserSession>;
++  fetchAuthSession: (options?: FetchAuthSessionOptions) => Promise<AuthSession>;
+```
+
+
```
Comment (israx) [inline src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx:251]: There is an extra space between the `Auth.currentSession` and `Auth.userSession` sections.

File: src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx (hunk 5 above)
Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx:238]: Good catch!

File: src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx (hunk 3 above)
Comment (erinleigh90) [reply src/pages/[platform]/build-a-backend/auth/auth-migration-guide/index.mdx:45]: Thanks! Missed that one

**Review by israx — DISMISSED** [review]
LGTM!

**Review by israx — DISMISSED** [review]

**Review by israx — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**